sda/cmd/verify/verify.go

```diff
@@ -0,0 +1,303 @@
+				infoErrorMessage := broker.InfoError{
+					Error:           "Message validation failed",
+					Reason:          err.Error(),
+					OriginalMessage: message,
+				}
+
+				body, _ := json.Marshal(infoErrorMessage)
+				if e := mq.SendMessage(delivered.CorrelationId, conf.Broker.Exchange, "error", body); e != nil {
+					log.Errorf("failed so publish message, reason: %v", err.Error())
+				}
+				if err := delivered.Ack(false); err != nil {
+					log.Errorf("Failed to Ack message, reason: %v", err.Error())
+				}
+
+				// Restart on new message
+				continue
+			}
+
+			// we unmarshal the message in the validation step so this is safe to do
+			_ = json.Unmarshal(delivered.Body, &message)
+
+			// If the file has been canceled by the uploader, don't spend time working on it.
+			status, err := db.GetFileStatus(delivered.CorrelationId)
+			if err != nil {
+				log.Errorf("failed to get file status, reason: %v", err.Error())
+				// Send the message to an error queue so it can be analyzed.
+				infoErrorMessage := broker.InfoError{
+					Error:           "Getheader failed",
+					Reason:          err.Error(),
+					OriginalMessage: message,
+				}
+
+				body, _ := json.Marshal(infoErrorMessage)
+				if e := mq.SendMessage(delivered.CorrelationId, conf.Broker.Exchange, "error", body); e != nil {
+					log.Errorf("failed so publish message, reason: %v", err.Error())
+				}
+
+				if err := delivered.Ack(false); err != nil {
+					log.Errorf("Failed acking canceled work, reason: %v", err.Error())
+				}
+
+				continue
+			}
+			if status == "disabled" {
+				log.Infof("file with correlation ID: %s is disabled, stopping verification", delivered.CorrelationId)
+				if err := delivered.Ack(false); err != nil {
+					log.Errorf("Failed acking canceled work, reason: %v", err.Error())
+				}
+
+				continue
+			}
+
+			header, err := db.GetHeader(message.FileID)
+			if err != nil {
+				log.Errorf("GetHeader failed for file with ID: %v, readon: %v", message.FileID, err.Error())
+
+				// Nack message so the server gets notified that something is wrong but don't requeue the message
+				if e := delivered.Nack(false, false); e != nil {
+					log.Errorf("Failed to nack following getheader error message")
+
+				}
+				// store full message info in case we want to fix the db entry and retry
+				infoErrorMessage := broker.InfoError{
+					Error:           "Getheader failed",
+					Reason:          err.Error(),
+					OriginalMessage: message,
+				}
+
+				body, _ := json.Marshal(infoErrorMessage)
+
+				// Send the message to an error queue so it can be analyzed.
+				if e := mq.SendMessage(delivered.CorrelationId, conf.Broker.Exchange, "error", body); e != nil {
+					log.Errorf("failed so publish message, reason: %v", err.Error())
+				}
+
+				continue
+			}
+
+			var file database.FileInfo
+
+			file.Size, err = archive.GetFileSize(message.ArchivePath)
+
+			if err != nil {
+				log.Errorf("Failed to get archived file size, reson: %v", err.Error())
```

Comment:
unless `err` contains some info, wouldn't it be useful if more info on the msg where this happened was printed?

Reply:
The error in this instance will be a storage related error either from `os.stat()` or `s3.Head()`. They are pretty verbose and concise.